src/main/java/com/shaft/api/RestActions.java

```diff
@@ -523,116 +253,376 @@ public static String getResponseXMLValue(Object response, String xmlPath) {
+        List<Object> responseBodyAttachment = new ArrayList<>();
+        if (responseBody != null) {
+            if (Boolean.TRUE.equals(isDiscrete)) {
+                try {
+                    ReportManager.logDiscrete("API Response - REST Body:\n"
+                            + IOUtils.toString(parseBodyToJson(responseBody), StandardCharsets.UTF_8));
+                } catch (IOException e) {
+                    ReportManager.logDiscrete("API Response - REST Body:\n" + responseBody.asString());
+                }
+            } else {
+                responseBodyAttachment.add("API Response");
+                switch (identifyBodyObjectType(responseBody)) {
+                    case 1:
+                        // json
+                        responseBodyAttachment.add("JSON Body");
+                        break;
+                    case 2:
+                        // xml
+                        responseBodyAttachment.add("XML Body");
+                        break;
+                    case 3:
+                        // binary... probably
+                        responseBodyAttachment.add("Body");
+                        break;
+                    case 4:
+                        // I don't remember... may be binary
+                        responseBodyAttachment.add("Body");
+                        break;
+                    default:
+                        // unreachable code
+                        break;
+                }
+                responseBodyAttachment.add(parseBodyToJson(responseBody));
+                return responseBodyAttachment;
+            }
+        }
+        return null;
+    }
+
+    private static int identifyBodyObjectType(Object body) {
         JSONParser parser = new JSONParser();
-        List<Object> expectedJSONAttachment = null;
-
         try {
-            // parse actual JSON into Object or Array
             org.json.simple.JSONObject actualJsonObject = null;
             org.json.simple.JSONArray actualJsonArray = null;
-
-            var actualObject = parser.parse(response.asString());
-            if (actualObject instanceof org.json.simple.JSONObject) {
-                actualJsonObject = (org.json.simple.JSONObject) parser.parse(response.asString());
+            if (body.getClass().getName().toLowerCase().contains("restassured")) {
+                // if it's a string (OR ARRAY) response body
+                try {
+                    String bodyString = ((io.restassured.response.ResponseBody<?>) body).asString();
+                    if (!bodyString.isEmpty()) {
+                        actualJsonObject = (org.json.simple.JSONObject) parser.parse(bodyString);
+                    }
+                } catch (ClassCastException e) {
+                    String bodyString = ((io.restassured.response.ResponseBody<?>) body).asString();
+                    if (!bodyString.isEmpty()) {
+                        actualJsonArray = (org.json.simple.JSONArray) parser.parse(bodyString);
```

Comment:
Codacy found an issue: [Unnecessary use of fully qualified name 'org.json.simple.JSONArray' due to existing import 'org.json.simple.JSONArray'](https://app.codacy.com/manual/mohab.mohieeldeen/SHAFT_ENGINE/pullRequest?prid=5737302)